src/net_processing.cpp

```diff
@@ -663,6 +671,35 @@ bool ProcessMessage(CNode *pfrom, std::string strCommand, CDataStream &vRecv, in
         // This step done after final handshake
         CheckAndRequestExpeditedBlocks(pfrom);
     }
+    else if (strCommand == NetMsgType::XUPDATE)
+    {
+        CXVersionMessage xUpdate;
+        vRecv >> xUpdate;
+        // check for peer trying to change non-changeable key
+        for (auto entry : xUpdate.xmap)
+        {
+            auto iter = XVer::keytype.find(entry.first);
+            if (iter == XVer::keytype.end())
+            {
+                pfrom->fDisconnect = true;
```

Comment:
As discussed, maybe silently ignoring is the better option. Needs input from others I guess. I would prefer the silent ignore.